### PR TITLE
chore: treat ttl label as seconds (previously it was a unix ts)

### PR DESCRIPTION
**Pull request by tonsV2**
Previously the ttl label was treated as a unix timestamp. This PR treats it as plain seconds thus making it easier to specify that an instance should live for... Say 5 minutes, which would added as 300 seconds.

The above is possible because the pods now also has a creation timestamp label which can be used as offset.

https://github.com/dhis2-sre/im-manager/pull/153 needs to be merged before this PR

### CODE REVIEW

File: pkg/pod/ttlDestroyHandler.go [hunk 1]
```diff
@@ -27,28 +27,39 @@ func (t ttlDestroyHandler) Supports() string {
...
+		return false
 	}
-	ttlTime := time.Unix(ttlInt, 0)
-	loc, _ := time.LoadLocation("UTC")
-	utc := ttlTime.In(loc)
-	now := time.Now()
-	return utc.Before(now)
+
+	ttlTime := time.Unix(creationTimestampInt+ttlInt, 0).UTC()
+	return ttlTime.Before(time.Now())
```
Comment (teleivo) [inline pkg/pod/ttlDestroyHandler.go:64]: Calling `time.Now()` inside functions you want/should test makes testing hard. Time is a side-effect.

If I understand correctly the inspector is not a long-running process rather invoked at certain intervals.

So we could push calling `time.Now()` into main and pass it down. This way we can more easily test that the TTL feature works.

File: pkg/pod/ttlDestroyHandler.go [hunk 2]
```diff
@@ -27,28 +27,39 @@ func (t ttlDestroyHandler) Supports() string {
...
 		t.producer.Produce(TtlDestroy, payload)
-	} else {
-		log.Println("No TTL found")
 	}
+
 	return nil
 }
 
-func (t ttlDestroyHandler) ttlBeforeNow(ttl string) bool {
+func (t ttlDestroyHandler) ttlBeforeNow(creationTimestamp string, ttl string) bool {
```
[the remark this answers is not in the document]
Comment (tonsV2) [reply pkg/pod/ttlDestroyHandler.go]: Most of our other services come with a web server and can therefor serve their own documentation from the open api specification generated by swagger. However that's not the case here. I did a description to the function though

File: pkg/pod/ttlDestroyHandler.go (hunk 1 above)
Comment (tonsV2) [reply pkg/pod/ttlDestroyHandler.go:67]: Is it really that difficult? Just use Time.Now() in the test when defining the input?

File: pkg/pod/ttlDestroyHandler.go [hunk 3]
```diff
@@ -27,28 +27,42 @@ func (t ttlDestroyHandler) Supports() string {
...
 	}
+
 	return nil
 }
 
-func (t ttlDestroyHandler) ttlBeforeNow(ttl string) bool {
-	ttlInt, err := strconv.ParseInt(ttl, 10, 64)
+// ttlBeforeNow return if creation time + ttl < now
+// creationTimestampLabel is a unix timestamp
+// ttlLabel is seconds
```
Comment (teleivo) [inline pkg/pod/ttlDestroyHandler.go:52]:
```suggestion
// ttlLabel is the pods time-to-live in seconds.
```

File: pkg/pod/ttlDestroyHandler.go [hunk 4]
```diff
@@ -27,28 +27,42 @@ func (t ttlDestroyHandler) Supports() string {
...
-	} else {
-		log.Println("No TTL found")
 	}
+
 	return nil
 }
 
-func (t ttlDestroyHandler) ttlBeforeNow(ttl string) bool {
-	ttlInt, err := strconv.ParseInt(ttl, 10, 64)
+// ttlBeforeNow return if creation time + ttl < now
```
Comment (teleivo) [inline pkg/pod/ttlDestroyHandler.go:50]:
```suggestion
// ttlBeforeNow returns true if the pod has expired according to its time to live.
```

File: pkg/pod/ttlDestroyHandler.go (hunk 1 above)
Comment (teleivo) [reply pkg/pod/ttlDestroyHandler.go:67]: You are right when it comes to the before and after cases. You can set `creationTimestamp+ttl` to a value long before or after `time.Now()`. Hitting the exact `time.Now()` might not work or result in a flaky test as the time you get has nanoseconds precision. So achieving `creationTimestamp+ttl` = `time.Now() (called in ttlBeforeNow)` will be hard/impossible.

For me its generally better to treat time as an input like any other. This makes testing easier and more reliable. I'll leave the decision up to you 😋

**Review by teleivo — COMMENTED** [review]
Have you tried writing a test? Or is this tested using an integration test?

**Review by teleivo — APPROVED** [review]

File: pkg/pod/ttlDestroyHandler.go (hunk 1 above)
Comment (tonsV2) [reply pkg/pod/ttlDestroyHandler.go:67]: For the sake of complete I'll say the following: I think we should keep it as is for now. It's testable as is and refactoring would only complicate the code unnecessary. But it is a valid point that "now" is hard coded.

**Pull request merged**